### DISCUSSION TITLE
grub-probe: error: unknown filesystem.

### DISCUSSION THREAD

**Discussion by spflanze** [openzfs/zfs#13751] [Q&A]
My ZFS install procedure is based on this one:
https://openzfs.github.io/openzfs-docs/Getting%20Started/Ubuntu/Ubuntu%2022.04%20Root%20on%20ZFS.html
the customization I did to that procedure I documented can be downloaded here:
https://www.mediafire.com/file/2w8mdb96tbzslub/Ubuntu_22.04_Root_on_ZFS_Encryption.odt/file

This is the result I get in Step 5.1
 
```
root@ubuntu:~# grub-probe /boot
grub-probe: error: unknown filesystem.
root@ubuntu:~# ls /boot
config-5.15.0-25-generic  memtest86+.bin                vmlinuz
grub                      memtest86+.elf                vmlinuz.old
initrd.img                memtest86+_multiboot.bin
initrd.img.old            System.map-5.15.0-25-generic

```
One major thing I did different is in Step 2.3 I did not create a legacy partition for BIOS booting. My computer boots UEFI only. I do not foresee needing to boot the hard drives on a BIOS computer. Furthermore it is believe the presence of this partition caused trouble when I attempted to install Pop!_OS while this partition was present. This is the conclusion reached in this thread I started:
https://www.reddit.com/r/pop_os/comments/wh59zx/how_much_should_be_allocate_for_a_pop_s_boot/
I did not know if this difference has anything to do with this error. What else could be causing this error?

I attempted to ignore this error, but it kept showing up in other steps in the procedure, and caused some important steps to fail, 
  
I accidentally closed the SSH session I am doing this installation with. When I reconnected the error message I get in response to the command changed. It is now:
```
root@ubuntu:~# grub-probe /boot
grub-probe: error: failed to get canonical path of `/cow'.
```
I need to shut down the target computer I doing the installation one, and the main computer I make an SSH connect to it with. Then I need to start again where I left off in the morning.  Any recommendations on how to get the target computer in the same sate it was before SSH connection is welcome. I do not want to have to start over again.

**Comment by sjorge** [comment]
Did you ever resolve this? I hit the same error and I followed the guide for a single disk layout.
I did create both UEFI and legacy partitions as asked even though I am also only booting via UEFI. So I doubt that was the issue.

**Reply by sjorge** [reply]
OK I'm just blind or it's my dyslexia playing up... I missed this command:

```
zfs create -o mountpoint=/boot bpool/BOOT/ubuntu_$UUID
```
Twice! Spotted it on my 3rd try.

**Comment by erikschul** [comment]
I ran into this issue as well. Somehow the boot partition had not been mounted correctly, so the /mnt/boot was actually part of rpool, which GRUB then correctly rejected as an unknown filesystem. I think the command to use to test is `grub-probe /boot`, but I didn't try it.

I rebooted, mounted rpool, deleted /mnt/boot, then mounted boot. Perhaps the ordering is important?
